tests/test_ami.py

```diff
@@ -37,6 +39,65 @@ def test_query(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_err_ami(self):
+        factory = self.replay_flight_data("test_ami_not_found_err")
+        ami_id = 'ami-123f000eee1f9f654'
```

Comment:
this is actually a malformed id, instead of a valid one, to trigger against us-east-1's separate error behavior.

Reply:
so the long form ami's seem to return back a malformed response, short form ami's are returning as not found for me in every region

Reply:
```
aws ec2 describe-images --owner=self --image-ids ami-1234567890abcdef0 --region us-east-1

An error occurred (InvalidAMIID.Malformed) when calling the DescribeImages operation: Invalid id: "ami-1234567890abcdef0"
```

```
aws ec2 describe-images --owner=self --image-ids ami-ffffffff ami-11111111 --region us-east-1

An error occurred (InvalidAMIID.NotFound) when calling the DescribeImages operation: The image ids '[ami-11111111, ami-ffffffff]' do not exist
```

```
aws ec2 describe-images --owner=self --image-ids ami-ffffffff ami-111111111 --region us-west-2

An error occurred (InvalidAMIID.Malformed) when calling the DescribeImages operation: Invalid id: "ami-111111111"
```

```
aws ec2 describe-images --owner=self --image-ids ami-1234567890abcdef0 --region us-west-2

An error occurred (InvalidAMIID.Malformed) when calling the DescribeImages operation: Invalid id: "ami-1234567890abcdef0"
```

long form image name: https://aws.amazon.com/about-aws/whats-new/2018/02/longer-format-resource-ids-are-now-available-in-amazon-ec2/

Reply:
Malformed is because it’s not per se a valid ami is, there’s some internal logic to the generation of the id that’s validated on api usage